src/utils/jsonp/jsonp-fetch.ts

```diff
@@ -0,0 +1,35 @@
+import { generateId } from 'utils/generate-id';
+import { registerJsonpResolver, unregisterJsonpResolver } from './jsonp-resolvers';
+
+export function jsonpFetch<DataType>(path: string, abortSignal?: AbortSignal): Promise<DataType> {
+    const script = document.createElement('script');
+    const callbackId = generateId(path);
+    let onAbort: undefined|EventListenerOrEventListenerObject;
+    return new Promise((resolve, reject) => {
```

Comment:
Could you please explain your motivation to use this approach to load data? Have you considered using Fetch API?

Reply:
As far as I understand. Fetch api does not support JSONP out of the box (because of CORS when you open local html file).  I create this fetch-like wrapper in order to load jsonp files. It implements standard JSONP flow. Loading jsonp like it is javascript, and by implementing global jsonp loader function. Plus some logic to make this api fetch-like.

Reply:
Anyway we can get rid of this, by trying to use libs like [jsonp-fetch](https://www.npmjs.com/package/fetch-jsonp) instead.